R/fread.R

```diff
@@ -142,6 +148,136 @@ fread <- function(input="",file=NULL,text=NULL,cmd=NULL,sep="auto",sep2="auto",d
     }
     # whitespace at the beginning or end of na.strings is checked at C level and is an error there; test 1804
   }
+  if (yaml) {
+    # for tracking which YAML elements may be overridden by being declared explicitly
+    call_args = names(match.call())
+    if (!requireNamespace('yaml', quietly = TRUE))
+      stop("'data.table' relies on the package 'yaml' to parse the file header; please add this to your library with install.packages('yaml') and try again.") # nocov
+    if (is.character(skip))
+      warning("Combining a search string as 'skip' and reading a ",
+              "YAML header may not work as expected -- currently, ",
+              "reading will proceed to search for 'skip' from ",
+              "the beginning of the file, NOT from the end of ",
+              "the metadata; please file an issue on GitHub if ",
+              "you'd like to see more intuitive behavior supported.")
+    # create connection to stream header lines from file:
+    #   https://stackoverflow.com/questions/9871307
+    f = base::file(input, 'r')
+    first_line = readLines(f, n=1L)
+    n_read = 1L
+    yaml_border_re = '^#?---'
+    if (!grepl(yaml_border_re, first_line)) {
+      close(f)
+      stop('Encountered <', substring(first_line, 1L, 50L),
+           if (nchar(first_line) > 50L) '...', '> at the first ',
+           'unskipped line (', 1L+skip, '), which does not ',
+           'constitute the start to a valid YAML header ',
+           '(expecting something matching regex "', yaml_border_re,
+           '"); please check your input and try again.')
+    }
+
+    yaml_comment_re = '^#'
+    yaml_string = character(0L)
+    while (TRUE) {
+      this_line = readLines(f, n=1L)
```

Comment:
what if there is only yaml header in the file and it is not completed by `----`? this line can thrown confusing warning on end of file with no new line

Reply:
See `inst/tests/csvy/test_incomplete_header.csvy` and test `1985.16`